### ISSUE TITLE
"undefined" error after clicking the "favourite" icon for a course

### ISSUE THREAD

**Issue by chrismyers81** [thepurpleblob/moodle-block_course_overview#10]
We're using Moodle 3.3.3+ with the Boost theme.

When our users click the start icon to "favorite" a course, they repeatedly get a popup saying simply "undefined", and keep getting that error message until they close their browser and reopen it.
![screenshot_20180119_120542](https://user-images.githubusercontent.com/5794323/35165230-218a19d2-fd13-11e7-8f9a-9f37f8bf832a.png)
I cranked up the Moodle debugging to "developer" and it didn't help much:
![screenshot_20180119_121016](https://user-images.githubusercontent.com/5794323/35165249-3993591c-fd13-11e7-84e6-7529e3f6a0a7.png)
The Apache error logs don't help much either:

> [Fri Jan 19 12:10:33.703731 2018] [:error] [pid 28932] [client 172.16.5.202:54466] Default exception handler: Coding error detected, it must be fixed by a programmer: Invalid json in request: Syntax error Debug: \nError code: codingerror\n* line 42 of /lib/ajax/service.php: coding_exception thrown\n* line 31 of /lib/ajax/service-nologin.php: call to require_once()\n
> [Fri Jan 19 12:10:33.719044 2018] [:error] [pid 28058] [client 172.16.5.202:54732] Default exception handler: Coding error detected, it must be fixed by a programmer: Invalid json in request: Syntax error Debug: \nError code: codingerror\n* line 42 of /lib/ajax/service.php: coding_exception thrown\n
> [Fri Jan 19 12:10:33.725943 2018] [:error] [pid 28009] [client 172.16.5.202:54366] Default exception handler: Coding error detected, it must be fixed by a programmer: Invalid json in request: Syntax error Debug: \nError code: codingerror\n* line 42 of /lib/ajax/service.php: coding_exception thrown\n
> [Fri Jan 19 12:10:33.847881 2018] [:error] [pid 30372] [client 172.16.5.202:54734] Default exception handler: Coding error detected, it must be fixed by a programmer: Invalid json in request: Syntax error Debug: \nError code: codingerror\n* line 42 of /lib/ajax/service.php: coding_exception thrown\n
> [Fri Jan 19 12:10:39.351886 2018] [:error] [pid 28009] [client 172.16.5.202:54366] Default exception handler: Coding error detected, it must be fixed by a programmer: Invalid json in request: Syntax error Debug: \nError code: codingerror\n* line 42 of /lib/ajax/service.php: coding_exception thrown\n, referer: http://moodle.millikin.edu/

Watching the Firefox developer tools, it simply says:
SyntaxError: JSON.parse: unexpected end of data at line 1 column 1 of the JSON data
![screenshot_20180119_122233](https://user-images.githubusercontent.com/5794323/35165311-6fd569de-fd13-11e7-858c-a19e4929955d.png)

I'll attach the .har file for you, encased in a .zip.
[Archive 18-01-19 12-18-50.har.zip](https://github.com/thepurpleblob/moodle-block_course_overview/files/1647523/Archive.18-01-19.12-18-50.har.zip)

**Comment by thepurpleblob** [comment]
I'm unable to reproduce this problem. The messages all seem to about json and we don't use any json calls in the code for this block so my suspicion is this is something else causing this.

**Comment by chrismyers81** [comment]
It's just curious to me that for us the problem reliably occurs every time a course is favorited or unfavorited. I'll attach a screen recording for you: https://screencast-o-matic.com/watch/cFVU29ov6d

**Comment by thepurpleblob** [comment]
Yes, but... it just seems to be you. The interesting thing from your screen recording is that the Font Awesome star used for favourite is getting corrupted when you click it. Have you run this with the Javascript console open? Is this "vanilla" Boost theme with no changes?

**Comment by chrismyers81** [comment]
Yep, this is vanilla Boost with no changes.

I'll build a new video with the console open here in a few minutes.

**Comment by thepurpleblob** [comment]
You really only need to check if any errors are displayed in the console. Otherwise, I'm struggling to think what might be going on

**Comment by chrismyers81** [comment]
https://screencast-o-matic.com/watch/cFVUoxovIl

It might be a bug in Boost, I can't say for sure. I can open a tracker issue with Moodle itself if needed, but we've only been hitting this problem after installing the legacy dashboard.

**Comment by chrismyers81** [comment]
Also, it looks like the Font Awesome star issue only happens when the URL contains "?unfavourite=####" or "?favourite=####" in  it. As soon as I revert back to the ".../my/" the icons look normal.

**Comment by thepurpleblob** [comment]
No errors in the console then. No errors for me in Boost with 3.3.3 and no other reports. 

Any chance I can get a log on that demonstrates this? You can PM me at howardsmiller@gmail.com

**Comment by chrismyers81** [comment]
I will check with our Ed Tech folks to try to get their approval, and will let you know.

**Comment by chrismyers81** [comment]
I just sent you an email with your account information.

**Comment by chrismyers81** [comment]
The interesting thing  is...after I upgraded my laptop from Firefox 57 to Firefox 58, I don't seem to experience the issue anymore. So maybe it's a browser-specific issue?

**Comment by chrismyers81** [comment]
I just checked on Firefox Developer Edition (v59) and it also isn't experiencing the issue anymore.

**Comment by chrismyers81** [comment]
Hmm, but another computer with FF 58 is experiencing the problem still.

**Comment by thepurpleblob** [comment]
I got an email starting, "We just wanted to let you know that your Moodle dev account has been created...", but no account details and no other email.

**Comment by chrismyers81** [comment]
Ha nice, I told Moodle to generate and email you the password. I'll send it along shortly.

**Comment by thepurpleblob** [comment]
I'm closing this as I have been unable to reproduce it.
